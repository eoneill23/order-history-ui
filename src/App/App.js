import React, { Component } from 'react';
import OrderList from '../OrderList/OrderList.js';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      orders: []
    }
  }

  componentDidMount() {

  }

  render() {
    return (
      <>
      <header></header>
      <main>
        <OrderList orders={this.state.orders}/>
      </main>
      </>
    )
  }
}

export default App;
