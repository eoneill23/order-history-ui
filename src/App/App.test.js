import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
 let wrapper, mockOrder, mockOrder2;

  beforeEach(() => {
    wrapper = shallow(<App />)
    mockOrder = { id: 1, img: 'google.com', name: 'Television', price: '200' }
    mockOrder2 = { id: 2, img: 'facebook.com', name: 'Radio', price: '100' }
  });

  it('should render without crashing', () => {
    expect(wrapper.length).toBe(1)
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should add an order when addOrder is invoked', () => {
    const expected = [mockOrder]
    expect(wrapper.state('orders')).toEqual([]);

    wrapper.instance().addOrder(mockOrder);

    expect(wrapper.state('orders')).toEqual(expected)
  });

  it('should delete an order when deleteOrder is invoked', () => {
    const expected = [mockOrder2];
    expect(wrapper.state('orders')).toEqual([]);

    wrapper.instance().addOrder(mockOrder);
    wrapper.instance().addOrder(mockOrder2);

    wrapper.instance().deleteOrder(1);

    expect(wrapper.state('orders')).toEqual(expected);
  });
});

