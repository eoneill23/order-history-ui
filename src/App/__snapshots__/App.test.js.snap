// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<Fragment>
  <header>
    <h1>
      My order history
    </h1>
    <OrderForm
      addOrder={[Function]}
    />
  </header>
  <main
    className="purchase-container"
  >
    <OrderList
      deleteOrder={[Function]}
      orders={Array []}
    />
  </main>
</Fragment>
`;
